restdocs-api-spec-openapi-generator/src/main/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20Generator.kt

```diff
@@ -267,16 +267,18 @@ object OpenApi20Generator {
             produces = modelsWithSamePathAndMethod.map { it.response.contentType }.distinct().filterNotNull().nullIfEmpty()
             parameters =
                     extractPathParameters(
-                        firstModelForPathAndMethod
+                        modelsWithSamePathAndMethod
                     ).plus(
-                        firstModelForPathAndMethod.request.requestParameters.map {
-                            requestParameterDescriptor2Parameter(
-                                it
-                            )
-                    }).plus(
-                        firstModelForPathAndMethod.request.headers.map {
-                            header2Parameter(it)
-                        }
+                        modelsWithSamePathAndMethod
+                                .flatMap { it.request.requestParameters }
+                                .distinct()
```

Comment:
I wonder if we should go for `distinctBy { it.name }` instead. If a resource would be documented with with parameters with the same name but e.g. different discription  `distinct()` would not kick in.

Reply:
Yes that makes sense. I changed the code & testcase accordingly.